File: game-logic-core/src/main/scala/fwc/gameLoading/TidesOfBattleCard.scala
package fwc.gameLoading

import fwc.JsonSerializable
import ujson.Value

case class TidesOfBattleCard(
                             code: Int, 
                             power: Int, 
                             death: Boolean = false, 
                             attack: Boolean = false, 
                             defense: Boolean = false
                           ) extends CardTrait with JsonSerializable{
  override def getCode: Int = code

  override def toJson: Value =
    val json = ujson.Obj(
      "code" -> code,
      "power" -> power,
    )
    if death
    then json.obj.addOne("death" -> death)
    if attack
    then json.obj.addOne("attack" -> attack)
    if defense
    then json.obj.addOne("defense" -> defense)
    json
}
       

File: bots/server_module/game_state/tides_of_battle_card.py
class TidesOfBattleCard:
    def __init__(
            self,
            code: int,
            power: int,
            death: bool = False,
            attack: bool = False,
            defense: bool = False
    ):
        self.code = code
        self.power = power
        self.death = death
        self.attack = attack
        self.defense = defense
